chapters/stack/intro/guides/stack-addressing/README.md

```diff
@@ -1,55 +1,51 @@
-# Stack operations
-The `stack_operations.asm` file demonstrates various stack operations.
-The main focus is to show how to manipulate the stack by pushing and popping values, and how to "allocate" and "deallocate" memory on the stack.
+# Stack addressing
+The `stack_addressing.asm` file demonstrates how data is stored on the stack, and especially in what order.
 
-> **Note:** Notice how `push` and `pop` are just syntactic sugar for the simpler `sub`, `add`, and `mov` instructions.
+Here's what an usual output for the compiled program would be:
+```
+0xff99fba8: 0xf7f46020
+0xff99fba4: 0xa
+0xff99fba0: 0xb
+0xff99fb9c: 0xc
+0xff99fb98: 0xd
+```
 
-For covenince, here's the contents of the file. To play around with it, download the lab locally.
+> **Note:** The last 4 values are the ones we pushed on stack.
+> What is the first one?
+
+> **Answer:** It is the old EBP we push at the start of the function.
```

Comment:
De ce old_ebp sub noi? E pusa de catre _start, sigur, dar gen -- de ce acolo? Acolo se termina stiva?

Reply:
Use English for everything so we stay consistent.

As for the explanation, the `main()` function from `stack-addressing.asm` pushes 5 values onto the stack:

```assembly
push ebp
mov ebp, esp

push dword 10
push dword 11
push dword 12
push dword 13
```
The first two instructions are there to set up `main()`'s stack frame.

The first value that is pushed is the old `ebp`. As for who sets it, it's more complicated. It's not that `_start` calls `main()`, but that `_start` calls `__libc_start_main()` which initialises the process's VAS and only then calls `main()`. You can find a more detailed explanation here [1]. So the value of the old `ebp` is the beginning of `__libc_start_main()`'s stack.

[1] http://dbp-consulting.com/tutorials/debugging/linuxProgramStartup.html